### PR TITLE
CA-205515 i18n: JA/SC: The error message about failed to join a domai…

### PR DESCRIPTION
**Pull request by minglumlu**
…n is not localized.

In this change, a logic is added to distinguish several errors which might occur during authentication of AD with pbis. Therefor, different api_error will be raised to make xenCenter being able to show different localised messages.

One new api_error is added in this change:
Api_errors.pool_auth_enable_failed_invalid_account["host";"message"]

### CODE REVIEW

File: ocaml/xapi/extauth_plugin_ADpbis.ml [hunk 1]
```diff
@@ -18,6 +18,36 @@
...
+      "LW_ERROR_DOMAIN_IS_OFFLINE", Auth_signature.E_UNAVAILABLE;
+      "LW_ERROR_INVALID_OU",       Auth_signature.E_INVALID_OU;
+      (* More errors to be caught here *)
+    ] 
+  in
+  let rec has_err lines err_pattern =
+    match lines with
+    | [] -> false
+    | line :: rest -> 
+      let split_to_words = fun str ->
```
Comment (mseri) [inline ocaml/xapi/extauth_plugin_ADpbis.ml]: I would move this function before the `match` call to have it upfront instead or redefining it for each element of the list

**Comment by mseri** [comment]
Only a minor comment. I am happy to see unit tests appearing, overall this seems a good improvement.

**Comment by minglumlu** [comment]
@mseri  Thanks for your comment.
I've moved the split_to_words before the match call.

File: ocaml/idl/datamodel.ml [hunk 2]
```diff
@@ -846,6 +846,8 @@ let _ =
     ~doc:"The host failed to enable external authentication." ();
   error Api_errors.auth_enable_failed_unavailable ["message"]
     ~doc:"The host failed to enable external authentication." ();
+  error Api_errors.auth_enable_failed_invalid_account["message"]
```
Comment (mseri) [inline ocaml/idl/datamodel.ml]: Can you add a space before `[`?

File: ocaml/idl/datamodel.ml [hunk 3]
```diff
@@ -899,6 +901,10 @@ let _ =
     ~doc:"The pool failed to enable external authentication." ();
   error Api_errors.pool_auth_enable_failed_duplicate_hostname ["host";"message"]
     ~doc:"The pool failed to enable external authentication." ();
+  error Api_errors.pool_auth_enable_failed_invalid_ou["host";"message"]
```
Comment (mseri) [inline ocaml/idl/datamodel.ml]: Can you add a space before `[`? (Also on line 906)

Comment (minglumlu) [inline ocaml/idl/datamodel.ml]: Oops. But it looks like no build error and runtime error.  Thank you @mseri

Comment (mseri) [inline ocaml/idl/datamodel.ml]: It's not an error, just a stylistic choice.

**Pull request merged**